cstr/src/lib.rs

```diff
@@ -68,16 +68,37 @@ pub unsafe fn input_text_to_string_w(text: *const WideChar, len: usize) -> Strin
     copy_nonoverlapping(text, dst.as_mut_ptr(), len);
     from_widechar_vec_lossy(dst)
 }
+///
+/// parse_attribute_string_a converts a null-separated utf8 doubly-null terminated cstring to a Rust
+/// string separated by `;`.
+///
+/// # Safety
+/// This converts a raw c-pointer to a Rust string, which requires unsafe operations
+///
+#[allow(clippy::uninit_vec)]
+pub unsafe fn parse_attribute_string_a(text: *const Char) -> String {
+    let mut dst = Vec::new();
+    let mut itr = text;
+    {
```

Comment:
This has the (slim) potential to access unitialized memory off the end of the cstring pointed to by `text: *const Char`. I think you should pass the length of the string to this function too, and add that `i + 1 < len` to the condition before the `offset` call. It makes the code less clean, so I'm sort of torn on this (since you'll have to keep track of i). I guess alternatively, you could create an end pointer as the `let end = itr.offset(len -1)`

Reply:
Unfortunately, we won't always know the length 😢 

For example, the list of attributes that are [passed in to ConfigDSN](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/configdsn-function?view=sql-server-ver16) don't include a length.

iODBC's `SQLGetPrivateProfileStringW` returns the keys of an entry, but does not return the length correctly in that it seems to not account for the null separators.

Reply:
Yea, I'm convinced this is the best we can do.